model/device.go

```diff
@@ -93,6 +93,136 @@ const (
 )
 
+// Consts containing defaults for all device configs.
+const (
+	defaultActPeriod int64 = 60
+	defaultMonPeriod int64 = 60
+)
+
+// Consts containing defaults for a controller config.
+const (
+	defaultControllerInputs  string = "A0,X50,X51,X60,X10"
+	defaultControllerOutputs string = "D14,D15,D16" // Peripheral Power.
+)
+
+// controllerDefaultVars is a map which relates controller variable names to their default values
+var controllerDefaultVars = map[string]string{
+	"AlarmNetwork":         "10",
+	"AlarmPeriod":          "5",     // 5 seconds.
+	"AlarmRecoveryVoltage": "840",   // 25.4545 V.
+	"AlarmVoltage":         "825",   // 25 V.
+	"AutoRestart":          "600",   // 10 minutes.
+	"Power":                "false", // initialise to OFF.
+	"Pulses":               "3",     // 3 pulses per cycle.
+	"PulseWidth":           "2",     // 1s ON, 1s OFF.
+	"PulseCycle":           "30",    // 30s between flashes.
+	"Suppress":             "false",
+}
+
+// controllerDefaultSensors contains a slice of the default sensors for a controller device.
+var controllerDefaultSensors = []SensorV2{
+	{Name: "Battery Voltage", Pin: "A0", Quantity: "DCV", Func: "scale", Args: "0.0289", Units: "V", Format: "round1"},
+	{Name: "Analog Value", Pin: "X10", Quantity: "OTH", Func: "none", Format: "round1"},
+	{Name: "Air Temperature", Pin: "X50", Quantity: "MWH", Func: "linear", Args: "0.1,-273.15", Units: "C", Format: "round1"},
+	{Name: "Humidity", Pin: "X51", Quantity: "MMB", Func: "scale", Args: "0.1", Units: "%", Format: "round2"},
+	{Name: "Water Temperature", Pin: "X60", Quantity: "MTW", Func: "linear", Args: "0.1,-273.15", Units: "C", Format: "round1"},
+}
+
+// controllerDefaultActs contains a slice of the default actuators for a controller device.
+var controllerDefaultActs = []ActuatorV2{
+	{Name: "Empty 1", Var: "Power1", Pin: "D16"},
+	{Name: "Empty 2", Var: "Power2", Pin: "D14"},
+	{Name: "Empty 3", Var: "Power3", Pin: "D15"},
+}
+
+// defaultVars contains a map between the device type and the default vars for that type.
+var defaultVars = map[string](map[string]string){
+	DevTypeController: controllerDefaultVars,
+}
+
+// defaultSensors contains a map between the device type and the default sensors for that type.
+var defaultSensors = map[string]([]SensorV2){
+	DevTypeController: controllerDefaultSensors,
+}
+
+// defaultActs contains a map between the device type and the default actuators for that type.
+var defaultActs = map[string]([]ActuatorV2){
+	DevTypeController: controllerDefaultActs,
+}
+
+// DeviceOption is a functional option to be supplied to NewDevice.
+type DeviceOption func(*Device) error
+
+// NewDevice creates a new device with the given parameters, and options.
+//
+// NOTE: Options are applied in order and any order dependent options should be
+// passed in order of execution.
+func NewDevice(ctx context.Context, skey int64, name, MAC string, settingsStore datastore.Store, options ...DeviceOption) (*Device, error) {
```

Comment:
Rename `settingsStore` to `store`, and make it the 2nd param for consistency with other model methods.